desc/objectives/_coils.py

```diff
@@ -1599,3 +1612,211 @@ def compute(self, params, constants=None):
         )
 
         return jnp.abs(link).sum(axis=0)
+
+
+class SurfaceCurrentRegularization(_Objective):
+    """Target the surface current magnitude.
+
+    compute::
+
+        w * ||K|| * ||e_theta x e_zeta||
+
+    where K is the winding surface current density, w is the
+    regularization parameter (the weight on this objective),
+    and ||e_theta x e_zeta|| is the magnitude of the surface normal i.e. the
+    surface jacobian ||e_theta x e_zeta||
+
+    This is intended to be used with a surface current::
+
+        K = n x ∇ Φ
+
+    i.e. a CurrentPotentialField
+
+    Intended to be used with a QuadraticFlux objective, to form
+    a problem similar to the REGCOIL algorithm described in [1]_ (if used with a
+    ``FourierCurrentPotentialField``, is equivalent to the ``simple``
+    regularization of the ``solve_regularized_surface_current`` method).
+
+    References
+    ----------
+    .. [1] Landreman, Matt. "An improved current potential method for fast computation
+      of stellarator coil shapes." Nuclear Fusion (2017).
+
+    Parameters
+    ----------
+    surface_current_field : CurrentPotentialField
+        Surface current which is producing the magnetic field, the parameters
+        of this will be optimized to minimize the objective.
+    source_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate current source at on
+        the winding surface. If used in conjunction with the QuadraticFlux objective,
+        with its ``field_grid`` matching this ``source_grid``, this replicates the
+        REGCOIL algorithm described in [1]_ .
+
+    """
+
+    weight_str = (
+        "weight : {float, ndarray}, optional"
+        "\n\tWeighting to apply to the Objective, relative to other Objectives."
+        "\n\tMust be broadcastable to to Objective.dim_f"
+        "\n\tWhen used with QuadraticFlux objective, this acts as the regularization"
+        "\n\tparameter (with w^2 = lambda), with 0 corresponding to no regularization."
+        "\n\tThe larger this parameter is, the less complex the surface current will "
+        "be,\n\tbut the worse the normal field."
+    )
+    __doc__ = __doc__.rstrip() + collect_docs(
+        target_default="``target=0``.",
+        bounds_default="``target=0``.",
+        loss_detail=" Note: has no effect for this objective.",
```

Comment:
why does the loss function not do anything?

Reply:
it will, that was a mistake